src/base/preferences.h

```diff
@@ -80,6 +81,13 @@ namespace DNS
 
 class SettingsStorage;
 
+class PreferencesParsingError : public std::runtime_error {
+    public:
```

Comment:
Should not be indented. Also the opening bracket above should be in it's own line.